biogeophys/FatesLeafBiophysParamsMod.F90

```diff
@@ -0,0 +1,301 @@
+         data=tmpreal)
+    allocate(lb_params%agross_btran_model(size(tmpreal,dim=1)))
+    call ArrayNint(tmpreal,lb_params%agross_btran_model)
+    deallocate(tmpreal)
+    
+    name = 'fates_leaf_stomatal_slope_medlyn'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%medlyn_slope)
+
+    name = 'fates_leaf_stomatal_slope_ballberry'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%bb_slope)
+
+    name = 'fates_leaf_stomatal_intercept'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%stomatal_intercept)
+
+    name = 'fates_maintresp_leaf_ryan1991_baserate'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%maintresp_leaf_ryan1991_baserate)
+
+    name = 'fates_maintresp_leaf_atkin2017_baserate'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%maintresp_leaf_atkin2017_baserate)
+
+    name = 'fates_maintresp_reduction_curvature'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%maintresp_reduction_curvature)
+
+    name = 'fates_maintresp_reduction_intercept'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%maintresp_reduction_intercept)
+
+    name = 'fates_maintresp_reduction_upthresh'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%maintresp_reduction_upthresh)
+
+    name = 'fates_leaf_vcmaxha'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%vcmaxha)
+
+    name = 'fates_leaf_jmaxha'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%jmaxha)
+
+    name = 'fates_leaf_vcmaxhd'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%vcmaxhd)
+
+    name = 'fates_leaf_jmaxhd'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%jmaxhd)
+
+    name = 'fates_leaf_vcmaxse'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%vcmaxse)
+
+    name = 'fates_leaf_jmaxse'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=lb_params%jmaxse)
+
+
+    return
+  end subroutine LeafBiophysReceiveParams
+
+  ! ====================================================================================
+  
+  ! subroutine LeafBiophysCheckParams()
+     !write (fates_log(),*)'error, incorrect stomatal slope scaling method (stomatal_intercept_method) applied'
+     !write (fates_log(),*)'valid options are:'
+     !write (fates_log(),*)'btran_on_gs = 0, 
+     !write (fates_log(),*)'btran_on_gs  = 1, apply btran to the stomatal conductance intercept'
+     !write (fates_log(),*)'btran_on_gs_gs1
+     !write (fates_log(),*)'you specified stomatal_intercept_method = ',stomatal_intercept_method
+     !call endrun(msg=errMsg(sourcefile, __LINE__))
+  ! end if
```

Comment:
Leave this in for future work?

Reply:
good question. Ill remove this.  We do check these params, but its done in EDPftVarcon.F90: FatesCheckParams().  The precedent is that modules that contain their own parameter definitions are checked all together in this routine.